### PR TITLE
add CVE-2019-20933 Detector 

### PR DESCRIPTION
**Pull request by secureness**
related to [this](https://github.com/google/tsunami-security-scanner-plugins/issues/279)

### CODE REVIEW

**Comment by secureness** [comment]
## vulnerable to CVE-2019-20933 
```yml
version: '2'
services:
 web:
   image: vulhub/influxdb:1.6.6
   environment:
    - INFLUXDB_HTTP_AUTH_ENABLED=true
    - INFLUXDB_ADMIN_USER=admin
    - INFLUXDB_ADMIN_PASSWORD=admin
    - INFLUXDB_DB=sample
   ports:
    - "8086:8086"
```
## vulnerable to missing authentication 
```yml
version: '2'
services:
 web:
   image: vulhub/influxdb:1.6.6
   environment:
    - INFLUXDB_HTTP_AUTH_ENABLED=false
    - INFLUXDB_DB=sample
   ports:
    - "8086:8086"
```

I'm gonna provide non-vulnerable influx container too.

**Comment by secureness** [comment]
Well, a little docker setup took me a lot of my time and now I understand why some open pull requests are still open!

# non-vulnerable influx container
`influxdb.conf` file 
```yml


reporting-disabled = false
bind-address = "0.0.0.0:8087"

[meta]
  dir = "/var/lib/influxdb/meta"
  retention-autocreate = true
  logging-enabled = true

[data]
  dir = "/var/lib/influxdb/data"
  index-version = "inmem"
  wal-dir = "/var/lib/influxdb/wal"
  wal-fsync-delay = "0s"
  validate-keys = false
  strict-error-handling = false
  query-log-enabled = true
  cache-max-memory-size = 1073741824
  cache-snapshot-memory-size = 26214400
  cache-snapshot-write-cold-duration = "10m0s"
  compact-full-write-cold-duration = "4h0m0s"
  compact-throughput = 50331648
  compact-throughput-burst = 50331648
  max-series-per-database = 1000000
  max-values-per-tag = 100000
  max-concurrent-compactions = 0
  max-index-log-file-size = 1048576
  series-id-set-cache-size = 100
  series-file-max-concurrent-snapshot-compactions = 0
  trace-logging-enabled = false
  tsm-use-madv-willneed = false

[coordinator]
  write-timeout = "10s"
  max-concurrent-queries = 0
  query-timeout = "0s"
  log-queries-after = "0s"
  max-select-point = 0
  max-select-series = 0
  max-select-buckets = 0

[retention]
  enabled = true
  check-interval = "30m0s"

[shard-precreation]
  enabled = true
  check-interval = "10m0s"
  advance-period = "30m0s"

[monitor]
  store-enabled = true
  store-database = "_internal"
  store-interval = "10s"

[subscriber]
  enabled = true
  http-timeout = "30s"
  insecure-skip-verify = false
  ca-certs = ""
  write-concurrency = 40
  write-buffer-size = 1000

[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = true
  log-enabled = true
  suppress-write-log = false
  write-tracing = false
  flux-enabled = false
  flux-log-enabled = false
  pprof-enabled = true
  pprof-auth-enabled = false
  debug-pprof-enabled = false
  ping-auth-enabled = true
  prom-read-auth-enabled = true
  https-enabled = false
  https-certificate = "/etc/ssl/influxdb.pem"
  https-private-key = ""
  max-row-limit = 0
  max-connection-limit = 0
  shared-secret = "4124124124ds5f4saf5sa45f5sa4f5saf54fsafdfsdf1241234"
  realm = "InfluxDB"
  unix-socket-enabled = false
  unix-socket-permissions = "0777"
  bind-socket = "/var/run/influxdb.sock"
  max-body-size = 25000000
  access-log-path = ""
  max-concurrent-write-limit = 0
  max-enqueued-write-limit = 0
  enqueued-write-timeout = 30000000000

[logging]
  format = "auto"
  level = "info"
  suppress-logo = false

[[graphite]]
  enabled = false
  bind-address = ":2003"
  database = "graphite"
  retention-policy = ""
  protocol = "tcp"
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "1s"
  consistency-level = "one"
  separator = "."
  udp-read-buffer = 0

[[collectd]]
  enabled = false
  bind-address = ":25826"
  database = "collectd"
  retention-policy = ""
  batch-size = 5000
  batch-pending = 10
  batch-timeout = "10s"
  read-buffer = 0
  typesdb = "/usr/share/collectd/types.db"
  security-level = "none"
  auth-file = "/etc/collectd/auth_file"
  parse-multivalue-plugin = "split"

[[opentsdb]]
  enabled = false
  bind-address = ":4242"
  database = "opentsdb"
  retention-policy = ""
  consistency-level = "one"
  tls-enabled = false
  certificate = "/etc/ssl/influxdb.pem"
  batch-size = 1000
  batch-pending = 5
  batch-timeout = "1s"
  log-point-errors = true

[[udp]]
  enabled = false
  bind-address = ":8089"
  database = "udp"
  retention-policy = ""
  batch-size = 5000
  batch-pending = 10
  read-buffer = 0
  batch-timeout = "1s"
  precision = ""

[continuous_queries]
  log-enabled = true
  enabled = true
  query-stats-enabled = false
  run-interval = "1s"

[tls]
  min-version = ""
  max-version = ""

```
```bash
docker run -e INFLUXDB_HTTP_AUTH_ENABLED=true   -e INFLUXDB_ADMIN_USER=admin -e INFLUXDB_ADMIN_PASSWORD=supersecretpassword  -e INFLUXDB_USER=telegraf -e INFLUXDB_USER_PASSWORD=secretpassword -p 8086:8086 -v $PWD/influxdb.conf:/etc/influxdb/influxdb.conf:ro influxdb:1.6.6 -config /etc/influxdb/influxdb.conf
```
just make sure the `$PWD/influxdb.conf` pointing to `influxdb.conf`

File: community/detectors/InfluxDB_cve_2019_20933/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# CVE-2019-20933 VulnDetector
+
+InfluxDB before 1.7.6 has an authentication bypass vulnerability in the authenticate function in
```
Comment (maoning) [inline community/detectors/InfluxDB_cve_2019_20933/README.md:3]: Could you also add a section about how InfluxDB doesn't enable authentication by default, and this plugin also checks for InfluxDB with no authentication.

Also, to conform with the directory naming convention, could you lowercase the detector folder name to "influxdb_cve_2019_20933"?

Comment (secureness) [reply community/detectors/InfluxDB_cve_2019_20933/README.md]: OK I improved the README and change the folder name to lowercase.

Comment (maoning) [reply community/detectors/InfluxDB_cve_2019_20933/README.md]: The update looks great, thanks for the fast turn-around. I've applied a few minor changes to your change set directly to pass the internal check. I simplified the db query a bit by dropping the "db=" part, it doesn't seem to be needed for the "show database" query after testing against the container images you provided.

**Comment by maoning** [comment]
Hi @secureness,

Your PR has been merged. This usually means a reward will be granted. Google will start the internal QC process and the reward amount will be determined based on the quality of the detector report. Please be patient and allow up to a week for the QC process to finish. You'll be notified once the decision is made.

Thanks!

**Comment by secureness** [comment]
It was a pleasure to work with this project and learn many new things.

**Comment by secureness** [comment]
> Hi @secureness,
> 
> Your PR has been merged. This usually means a reward will be granted. Google will start the internal QC process and the reward amount will be determined based on the quality of the detector report. Please be patient and allow up to a week for the QC process to finish. You'll be notified once the decision is made.
> 
> Thanks!

Hi @maoning
I just want to follow up the process as I'm worried now if there is some things wrong with the process or not.

**Comment by maoning** [comment]
@secureness everything looks good, the patch reward has been granted, you should be notified in the next day or so.

**Comment by secureness** [comment]
Hi again, I'm worried if there is a problem in the payment process, could you give me information about that?

**Pull request merged**